app/mailers/user_notifications.rb

```diff
@@ -231,14 +231,14 @@ def digest(user, opts = {})
       limit: SiteSetting.digest_topics + SiteSetting.digest_other_topics,
       top_order: true,
     }
-    topics_for_digest = Topic.for_digest(user, min_date, digest_opts).to_a
+    topics_for_digest = Topic.for_digest(user, min_date, digest_opts)
     if topics_for_digest.empty? && !user.user_option.try(:include_tl0_in_digests)
       # Find some topics from new users that are at least 24 hours old
       topics_for_digest =
-        Topic
-          .for_digest(user, min_date, digest_opts.merge(include_tl0: true))
-          .where("topics.created_at < ?", 24.hours.ago)
-          .to_a
+        Topic.for_digest(user, min_date, digest_opts.merge(include_tl0: true)).where(
+          "topics.created_at < ?",
+          24.hours.ago,
+        )
```

Comment:
Mostly formatting update. Relevant change is remove eager `#to_a` so we can use `#joins` later.